direct/io-blob/src/main/java/cz/o2/proxima/direct/blob/BlobStorageAccessor.java

```diff
@@ -56,26 +55,38 @@
   /** Maximal amount of time (in milliseconds) a partition containing multiple blobs can span. */
   public static final String PARTITION_MAX_TIME_SPAN_MS = "partition.max-time-span-ms";
 
-  final Map<String, Object> cfg;
+  final AttributeFamilyDescriptor familyDescriptor;
 
-  protected BlobStorageAccessor(EntityDescriptor entityDesc, URI uri, Map<String, Object> cfg) {
-    super(entityDesc, uri);
-    this.cfg = new HashMap<>(cfg);
+  transient Map<String, Object> cfg;
+
+  protected BlobStorageAccessor(AttributeFamilyDescriptor family) {
+    super(family.getEntity(), family.getStorageUri());
+    this.familyDescriptor = family;
   }
 
   public Map<String, Object> getCfg() {
+    if (cfg == null) {
+      /**
+       * Creating new HashMap because familyDescriptor return already unmodifiableMap and we'll not
+       * be able to override it from tests
+       */
+      cfg = new HashMap<>(familyDescriptor.getCfg());
+    }
     return Collections.unmodifiableMap(cfg);
   }
 
   @VisibleForTesting
   void setCfg(String key, Object value) {
+    getCfg();
```

Comment:
Is this OK? This introduces a somewhat unexpected contract:
```java
 accessor.setCfg(...);
 cloned = SerializableUtils.clone(accessor);
 cloned.getCfg(...); // the set configuration will be missing here
 ```

Reply:
Reverted back, I think fixing Serializable contract doesn't make sense here and we can probably remove it later.